src/core/mac/sub_mac.cpp

```diff
@@ -341,18 +345,26 @@ void SubMac::StartCsmaBackoff(void)
 #if OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
     if (mTransmitFrame.mInfo.mTxInfo.mPeriod != 0)
     {
-        uint32_t phaseNow     = (otPlatTimeGet() / kUsPerTenSymbols) % mTransmitFrame.mInfo.mTxInfo.mPeriod;
-        uint32_t phaseDesired = mTransmitFrame.mInfo.mTxInfo.mPhase;
-
         SetState(kStateCslTransmit);
 
-        if (phaseNow < phaseDesired)
-        {
-            mTimer.Start((phaseDesired - phaseNow) * kUsPerTenSymbols);
-        }
-        else if (phaseNow > phaseDesired)
+        if (ShouldHandleTransmitDelay())
```

Comment:
Is "target time" more descriptive?
```suggestion
        if (ShouldHandleTransmitTargetTime())
```